File: Java/LeetCode/src/main/java/com/example/sort/selectSort.java
package com.example.sort;

/**
 * @author huangchunchen
 * @date 2021/8/22 21:10
 * @description
 */
public class selectSort {
    public static void main(String[] args) {
        int[] nums = {5,1,8,3,6,2};
        selectSort(nums);
        System.out.println(nums);
    }

    public static void selectSort(int[] nums){
        int len = nums.length;
        for (int i=0;i<len;i++){
            int min = i;
            for (int j=i+1;j<len;j++){
                if (nums[min] > nums[j]){
                    min=j;
                }
            }
            swap(nums, i, min);
        }
    }

    public static void swap(int[] nums, int i, int j){
        int temp = nums[i];
        nums[i] = nums[j];
        nums[j] = temp;
    }
}

File: cdd/LeetCode/LeetCode/selectSort.cpp
#include <vector>
#include <stdio.h>

using namespace std;

class Solution {
public:
	void selectSort(vector<int>& nums)
	{
		int length = nums.size();
		for (int i = 0; i < length; i++)
		{
			int min = i;
			for (int j = i+1; j < length; j++)
			{
				if (nums[min] > nums[j])
				{
					min = j;
				}
			}
			swap(nums, min, i);
		}
	}

	void swap(vector<int>& nums, int i, int j)
	{
		int temp = nums[i];
		nums[i] = nums[j];
		nums[j] = temp;
	}
};